index.js

```diff
@@ -60,6 +60,11 @@ function register (server, options, next) {
     reply.continue()
   })
 
+  server.ext('onPreHandler', (request, reply) => {
+    request.logger = (request.logger || logger).child({ req: request })
```

Comment:
So I should add an option and add `payload: request.payload` like above if the option is truthy?